pkg/cloud/services/ec2/instances.go

```diff
@@ -636,6 +638,7 @@ func (s *Service) runInstance(role string, i *infrav1.Instance) (*infrav1.Instan
 	input.InstanceMarketOptions = getInstanceMarketOptionsRequest(i.SpotMarketOptions)
 	input.MetadataOptions = getInstanceMetadataOptionsRequest(i.InstanceMetadataOptions)
 	input.PrivateDnsNameOptions = getPrivateDNSNameOptionsRequest(i.PrivateDNSName)
+	input.CapacityReservationSpecification = getCapacityReservationSpecification(i.CapacityReservationID)
```

Comment:
what happens if this was changed day2? should call the modify-instance-capacity-reservation-attributes API on the EC2 instance?

Reply:
We can't modify the capacity-reservation attributes unless the instance is in a stopped state.

Reply:
Then should the webhook disallow changing the field's value?

Reply:
yes, if we try to modify the running instance capacity-reservation attributes it will raise an error `IncorrectInstanceState`.

Reply:
@athiruma I think the suggestion is to add something like https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1beta2/awscluster_webhook.go#L109 so that the user cannot modify the field if the instance is running.

Reply:
[https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6e43273dd65d166332e607bf0858aa6a55491650/api/v1beta2/awsmachine_webhook.go#L137-L139](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/6e43273dd65d166332e607bf0858aa6a55491650/api/v1beta2/awsmachine_webhook.go#L137-L139) It is not allowing to modify any values in AWSMachineSpec expect the values which are configured to modify. So no need to add any webhook for the CapacityReservationID.

cc: @AndiDog, @nrb